src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java

```diff
@@ -245,6 +245,7 @@ public void onResume() {
         }
     }
 
+    @SuppressFBWarnings(value = "SE_INNER_CLASS", justification = "Not sure if it is intended or if this inner class should be static.")
```

Comment:
Converting the class to inner static would be better but some refactoring is needed to achieve that.

Reply:
I was wondering about this one. 

The refactoring needed would be to take care of `KubernetesCloud cloud = resolveCloud();` which calls `resolveCloud` coming from the parent class.

I was wondering if that `PodTemplateCallback` was used in the Serialization somewhere?

I was thinking about either adding a `KubernetesCloud` object in its constructor so I would inject the `resolveCloud()` result when constructing the object, or maybe directly adding a `PodTemplateStepExecution` so I can call the `resolveCloud` method from there. But I don't want to mess up with whatever could be serialized here and since I am not too sure about it I thought I'd rather ask.

Any opinions on that?

Reply:
It implements Serializable so I would assume it can get serialized.

I recommend adding `cloudName` as a field and converting `resolveCloud` to a static method.

Reply:
Done